Sources/xcproj/PBXLegacyTarget.swift

```diff
@@ -0,0 +1,109 @@
+    public var buildWorkingDirectory: String?
+    
+    public init(reference: String,
+                name: String,
+                buildToolPath: String? = nil,
+                buildArgumentsString: String? = nil,
+                passBuildSettingsInEnvironment: Bool = false,
+                buildWorkingDirectory: String? = nil,
+                buildConfigurationList: String? = nil,
+                buildPhases: [String] = [],
+                buildRules: [String] = [],
+                dependencies: [String] = [],
+                productName: String? = nil,
+                productReference: String? = nil,
+                productType: PBXProductType? = nil) {
+        self.buildToolPath = buildToolPath
+        self.buildArgumentsString = buildArgumentsString
+        self.passBuildSettingsInEnvironment = passBuildSettingsInEnvironment
+        self.buildWorkingDirectory = buildWorkingDirectory
+        super.init(reference: reference, name: name, buildConfigurationList: buildConfigurationList, buildPhases: buildPhases, buildRules: buildRules, dependencies: dependencies, productName: productName, productReference: productReference, productType: productType)
+    }
+    
+    // MARK: - Decodable
+    
+    fileprivate enum CodingKeys: String, CodingKey {
+        case buildToolPath
+        case buildArgumentsString
+        case passBuildSettingsInEnvironment
+        case buildWorkingDirectory
+        case buildConfigurationList
+        case buildPhases
+        case buildRules
+        case dependencies
+        case name
+        case productName
+        case productReference
+        case productType
+        case reference
+    }
+    
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.buildToolPath = try container.decodeIfPresent(.buildToolPath)
+        self.buildArgumentsString = try container.decodeIfPresent(.buildArgumentsString)
+        let passBuildSettingsInEnvironmentString: String? = try container.decodeIfPresent(.passBuildSettingsInEnvironment)
+        self.passBuildSettingsInEnvironment = passBuildSettingsInEnvironmentString.flatMap(Int.init).map{ $0 == 1 } ?? false
+        self.buildWorkingDirectory = try container.decodeIfPresent(.buildWorkingDirectory)
+        try super.init(from: decoder)
+    }
+    
+    public static func == (lhs: PBXLegacyTarget,
+                           rhs: PBXLegacyTarget) -> Bool {
+        return (lhs as PBXTarget) == (rhs as PBXTarget) &&
+            lhs.buildToolPath == rhs.buildToolPath &&
+            lhs.buildArgumentsString == rhs.buildArgumentsString &&
+            lhs.passBuildSettingsInEnvironment == rhs.passBuildSettingsInEnvironment &&
+            lhs.buildWorkingDirectory == rhs.buildWorkingDirectory
+    }
+    
+    override func plistValues(proj: PBXProj, isa: String) -> (key: CommentedString, value: PlistValue) {
```

Comment:
We didn't use to test this object method but we started to do it recently because it's usually the source of inconsistencies when saving the project back *(e.g. bad comment, missing comment)*. `PlistValue` [is already equatable](https://github.com/xcodeswift/xcproj/blob/master/Sources/xcproj/PlistValue.swift#L74) so it should facilitate writing the test. I think it's important to test that the values have the proper comment and that are written with the proper key.

Reply:
If I'm understanding this right: You want me to confirm that the Plist comments are the same as what would exist if this target were created from Xcode? I just confirmed that all these new fields do not have any Plist comments.

Or are you saying that I should write a unit test somewhere? Can you point me to an example of a plistValues one that already exists (I can't seem to find it).

Reply:
I meant more your second point. You can check out [these tests](https://github.com/xcodeswift/xcproj/blob/master/Tests/xcprojTests/PBXSourceTreeSpec.swift#L36) where we test that the returned `plistValues` are the expected ones.